async_upnp_client/client_factory.py

```diff
@@ -386,7 +386,12 @@ async def _async_get(self, url: str) -> ET.Element:
         if status_code != 200:
             raise UpnpResponseError(status=status_code, headers=response_headers)
 
-        description: str = (response_body or "").rstrip(" \t\r\n\0")
+        """Create a fake service if response lenght is zero to avoid parse errors."""
```

Comment:
The `_async_get` function is called for getting the device description too. In this case it would be better to handle the error in the calling function (`_async_create_service`).

Reply:
Perhaps replace [line 146](https://github.com/StevenLooman/async_upnp_client/pull/127/files#diff-22fd5c7b741245aa146d9f21537a681671dec9e1a2d70b463a8a14467138ad5fL146) above with:
```py
        try:
            scpd_el = await self._async_get(scpd_url)
        except UpnpXmlParseError:
            if not self._non_strict:
                raise
            _LOGGER.debug("Ignoring bad XML document from URL %s", scpd_url)
            scpd_el = ET.Element(f"{{{NS['service']}}}scpd")
```

This way we'll catch the error only in the function that cares, and use an empty XML object instead. Kind of like parsing the fake response here, but skipping the parser step.